Samples/excel-add-in-shape-dashboard/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,431 @@
+    maxSale.fill.foregroundColor = "#FF6600";
+    maxSale.textFrame.textRange.text = text6.values.toString() + "\nMax Sale";
+
+    let text7 = context.workbook.worksheets.getItem("Sample").getRange("E156:E156");
+    text7.load("values");
+    await context.sync();
+    let sumSale = shapeGroup.items[7];
+    sumSale.fill.foregroundColor = "#FF6600";
+    sumSale.textFrame.textRange.text = text7.values.toString() + "\nSum Sale";
+
+    await context.sync();
+    showStatus("Add information to dashboard - success!", false);
+  });
+}
+
+async function changeFontFormat() {
+  await Excel.run(async (context) => {
+    showStatus("Running...", false);
+    let shapes = context.workbook.worksheets.getItem("Sample").shapes;
+    shapes.load("items");
+    await context.sync();
+
+    let shapeGroup = shapes.items[0].group.shapes;
+    shapes.items[0].onActivated.add(() => tryCatch(onActivate));
+    shapes.items[0].onDeactivated.add(() => tryCatch(onDeactivate));
+    shapeGroup.load("items");
+    await context.sync();
+
+    for (let i = 0; i < shapeGroup.items.length; i++) {
+      let shp = shapeGroup.items[i];
+      shp.textFrame.textRange.font.name = "Consolas";
+
+      shp.textFrame.verticalAlignment = Excel.ShapeTextVerticalAlignment.middle;
+      shp.textFrame.horizontalAlignment = Excel.HorizontalAlignment.center;
+    }
+
+    let region = shapeGroup.items[0];
+    region.textFrame.textRange.getSubstring(0, 1).font.size = 30;
+    region.textFrame.textRange.getSubstring(1).font.size = 17;
+    region.textFrame.textRange.getSubstring(1).font.color = "#FFFFCC";
+
+    let state = shapeGroup.items[1];
+    state.textFrame.textRange.getSubstring(0, 2).font.size = 30;
+    state.textFrame.textRange.getSubstring(2).font.size = 17;
+    state.textFrame.textRange.getSubstring(2).font.color = "#FFFFCC";
+
+    let category = shapeGroup.items[3];
+    category.textFrame.textRange.getSubstring(0, 1).font.size = 30;
+    category.textFrame.textRange.getSubstring(1).font.size = 12;
+    category.textFrame.textRange.getSubstring(1).font.color = "#99CCFF";
+
+    let subCategory = shapeGroup.items[4];
+    subCategory.textFrame.textRange.getSubstring(0, 2).font.size = 30;
+    subCategory.textFrame.textRange.getSubstring(2).font.size = 12;
+    subCategory.textFrame.textRange.getSubstring(2).font.color = "#99CCFF";
+
+    let maxSale = shapeGroup.items[6];
+    maxSale.textFrame.textRange.getSubstring(0, 8).font.size = 18;
+    maxSale.textFrame.textRange.getSubstring(8).font.size = 13;
+    maxSale.textFrame.textRange.getSubstring(8).font.color = "FFFF66";
+
+    let sumSale = shapeGroup.items[7];
+    sumSale.textFrame.textRange.getSubstring(0, 9).font.size = 15;
+    sumSale.textFrame.textRange.getSubstring(9).font.size = 13;
+    sumSale.textFrame.textRange.getSubstring(9).font.color = "FFFF66";
+
+    await context.sync();
+    showStatus("Change information format - success!", false);
+  });
+}
+
+async function onActivate() {
```

Comment:
This method adds a lot of text and formatting that immediately gets replaced. It's very confusing.

Reply:
Added a button for adding event handler to it.